### ISSUE TITLE
Add `setup` and` emits` to the options checked by the `vue/order-in-components` rule.

### ISSUE THREAD

**Issue by ota-meshi** [vuejs/eslint-plugin-vue#1134]
**Please describe what the rule should do:**

The `vue/order-in-components` rule checks `setup` and `emits` options. (`setup` and `emits`　added in Vue.js 3.x)
The default order for the `vue/order-in-components` rule is designed according to the [style guide].
But, `setup` and `emits` are new options added in Vue.js 3.x, and the order is not defined in the [style guide].
We first need to define the [style guide].

**What category should the rule belong to?**

- [X] Enforces code style
- [ ] Warns about a potential error
- [ ] Suggests an alternate way of doing something
- [ ] Other (please specify:)

**Provide 2-3 code examples that this rule should warn about:**

```vue
TODO
```

**Additional context**

- https://vuejs.org/v2/style-guide/#Component-instance-options-order-recommended
- https://github.com/vuejs/eslint-plugin-vue/issues/1035

[style guide]: https://vuejs.org/v2/style-guide/#Component-instance-options-order-recommended

**Comment by ota-meshi** [comment]
Lifecycle hooks may also have changed...

**Comment by ota-meshi** [comment]
We also need to add `provide` / `inject`, which already exists from Vue 2.x.

**Comment by ota-meshi** [comment]
The internal style guide documentation has been updated, so I close this issue.
